Private/tkrupa/cache encryption docs (#28035)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/running-inference/optimize-inference/optimizing-latency/model-caching-overview.rst

Set "cache_encryption_callbacks" config option to enable cache encryption
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

> If model caching is enabled in CPU Plugin, the model topology can be encrypted when saving to the cache and decrypted when loading from the cache. This property can currently be set only in ``compile_model``.

Reviewer: ```suggestion
If model caching is enabled in the CPU Plugin, the model topology can be encrypted while it is saved to the cache and decrypted when it is loaded from the cache. This property can currently be set only in ``compile_model``.
```

         :language: cpp
         :fragment: [ov:caching:part4]

> If model caching is enabled in GPU Plugin, everything can be encrypted when saving to the cache and decrypted when loading from the cache. Full encryption only works when ``CacheMode`` property is set to ``OPTIMIZE_SIZE``.

Reviewer: ```suggestion
If model caching is enabled in the GPU Plugin, the model topology can be encrypted while it is saved to the cache and decrypted when it is loaded from the cache. Full encryption only works when the ``CacheMode`` property is set to ``OPTIMIZE_SIZE``.
```
